src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
+      children.push(val[j])
+    }
   }
+  else {
+    children.push(typeof val === 'number' ? val + '' : val)
+  }
+}
 
-  while (stack.length) {
-    if (Array.isArray((node = stack.pop()))) {
-      for (var i = node.length; i--; ) {
-        stack[stack.length] = node[i]
+export function h(tag, data, values) {
+  var children = []
+
+  if (arguments.length > 2) {
```

Comment:
I'll check this, but from what I remember, accessing an out of bounds array deopts the function because the VM has to do additional bounding checks.